occt-import-js/src/js-interface.cpp

```diff
@@ -73,19 +73,19 @@ class HierarchyWriter
                     indexArr.set (triangleCount * 3 + 2, vertexOffset + v2);
                     triangleCount += 1;
                 });
+                emscripten::val faceColorObj (emscripten::val::object ());
+                faceColorObj.set ("first", triangleOffset);
+                faceColorObj.set ("last", triangleCount - 1);
                 Color faceColor;
                 if (face.GetColor (faceColor)) {
-                    emscripten::val faceColorObj (emscripten::val::object ());
-                    faceColorObj.set ("first", triangleOffset);
-                    faceColorObj.set ("last", triangleCount - 1);
                     emscripten::val colorArr (emscripten::val::array ());
                     colorArr.set (0, faceColor.r);
                     colorArr.set (1, faceColor.g);
                     colorArr.set (2, faceColor.b);
                     faceColorObj.set ("color", colorArr);
-                    faceColorArr.set (faceColorCount, faceColorObj);
-                    faceColorCount += 1;
                 }
+                faceColorArr.set (faceColorCount, faceColorObj);
+                faceColorCount += 1;
```

Comment:
It can also be called simply `faceCount`. It will always have values so the `if (faceColorCount > 0)` is not needed a couple of lines below.